### PR TITLE
More Sinsemilla optimisations

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

**Comment by str4d** [comment]
Overall PR vs current `main`:
```
Sinsemilla/hash-to-point/510                                                                            
                        time:   [90.795 us 91.082 us 91.385 us]
                        change: [-11.178% -8.9634% -7.1617%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 5 outliers among 100 measurements (5.00%)
  2 (2.00%) high mild
  3 (3.00%) high severe
Sinsemilla/hash/510     time:   [99.558 us 100.12 us 100.89 us]                                
                        change: [-7.0946% -6.2151% -5.3430%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 7 outliers among 100 measurements (7.00%)
  5 (5.00%) high mild
  2 (2.00%) high severe
Sinsemilla/commit/510   time:   [164.09 us 164.61 us 165.18 us]                                  
                        change: [-34.102% -33.546% -32.942%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 8 outliers among 100 measurements (8.00%)
  2 (2.00%) high mild
  6 (6.00%) high severe
Sinsemilla/short-commit/510                                                                            
                        time:   [164.46 us 165.14 us 165.84 us]
                        change: [-35.744% -34.436% -33.488%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 4 outliers among 100 measurements (4.00%)
  4 (4.00%) high mild
Sinsemilla/hash-to-point/520                                                                            
                        time:   [92.807 us 93.099 us 93.430 us]
                        change: [-6.9973% -6.2792% -5.5196%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 6 outliers among 100 measurements (6.00%)
  2 (2.00%) high mild
  4 (4.00%) high severe
Sinsemilla/hash/520     time:   [101.59 us 102.04 us 102.51 us]                                
                        change: [-7.5361% -6.7735% -6.0011%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 3 outliers among 100 measurements (3.00%)
  1 (1.00%) high mild
  2 (2.00%) high severe
Sinsemilla/commit/520   time:   [165.01 us 165.50 us 166.06 us]                                  
                        change: [-33.475% -32.998% -32.488%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 5 outliers among 100 measurements (5.00%)
  3 (3.00%) high mild
  2 (2.00%) high severe
Sinsemilla/short-commit/520                                                                            
                        time:   [166.15 us 167.03 us 168.17 us]
                        change: [-33.840% -33.014% -32.195%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 8 outliers among 100 measurements (8.00%)
  2 (2.00%) high mild
  6 (6.00%) high severe
Sinsemilla/hash-to-point/1086                                                                            
                        time:   [195.33 us 196.15 us 197.01 us]
                        change: [-7.2010% -6.5700% -5.8896%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 3 outliers among 100 measurements (3.00%)
  2 (2.00%) high mild
  1 (1.00%) high severe
Sinsemilla/hash/1086    time:   [202.64 us 203.58 us 204.70 us]                                 
                        change: [-7.6129% -6.7820% -5.9938%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 5 outliers among 100 measurements (5.00%)
  2 (2.00%) high mild
  3 (3.00%) high severe
Sinsemilla/commit/1086  time:   [269.18 us 271.95 us 275.29 us]                                   
                        change: [-26.349% -25.477% -24.646%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  5 (5.00%) high mild
  4 (4.00%) high severe
Sinsemilla/short-commit/1086                                                                            
                        time:   [267.06 us 267.95 us 268.96 us]
                        change: [-29.794% -27.538% -25.551%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 7 outliers among 100 measurements (7.00%)
  3 (3.00%) high mild
  4 (4.00%) high severe
```

File: src/primitives/sinsemilla/addition.rs [hunk 1]
```diff
@@ -53,3 +53,28 @@ impl Add<pallas::Point> for IncompletePoint {
...
     }
 }
+
+impl Add<pallas::Affine> for IncompletePoint {
+    type Output = IncompletePoint;
+
+    /// Specialisation of incomplete addition for mixed addition.
+    fn add(self, rhs: pallas::Affine) -> Self::Output {
+        // ⊥ ⊹ ⊥ = ⊥
+        // ⊥ ⊹ P = ⊥
```
Comment (daira) [inline src/primitives/sinsemilla/addition.rs:63]:
```suggestion
        // ⊥ ⸭ ⊥ = ⊥
        // ⊥ ⸭ P = ⊥
```

File: src/primitives/sinsemilla/addition.rs [hunk 2]
```diff
@@ -53,3 +53,28 @@ impl Add<pallas::Point> for IncompletePoint {
         self + IncompletePoint(CtOption::new(rhs, 1.into()))
     }
 }
+
+impl Add<pallas::Affine> for IncompletePoint {
+    type Output = IncompletePoint;
+
+    /// Specialisation of incomplete addition for mixed addition.
+    fn add(self, rhs: pallas::Affine) -> Self::Output {
```
Comment (daira) [inline src/primitives/sinsemilla/addition.rs:61]:
```suggestion
    #[allow(clippy::suspicious_arithmetic_impl)]
    fn add(self, rhs: pallas::Affine) -> Self::Output {
```

**Review by daira — APPROVED** [review]
utACK with suggestions.

**Pull request merged**